### DISCUSSION TITLE
mpd error while loading shared libraries: libicuuc.so.67

### DISCUSSION THREAD

**Discussion by lin-silas** [MusicPlayerDaemon/MPD#1374] [General]
Hi,

I've disabled icu at meson config option, but MPD still wants to load shared libraries at runtime.
How to prevent MPD loading unused libraries?

**Comment by MaxKellermann** [comment]
Why do you believe disabling ICU should prevent MPD from loading any shared libraries at runtime?
I think you mean something else, but your question is so sparse on details, it's impossible to help or even guess what you're really after.

**Comment by lin-silas** [comment]
Thanks for your rapid apply.

This is my config

```
meson configure output/release \
  -Dsysconfdir='/etc' \
  -Dcue=true \
  -Ddaemon=true \
  -Ddatabase=true \
  -Depoll=true \
  -Deventfd=true \
  -Dfifo=true \
  -Dinotify=true \
  -Dlocal_socket=true \
  -Dpipe=true \
  -Dsignalfd=true \
  -Dtcp=true \
  -Dalsa=enabled \
  -Dffmpeg=enabled \
  -Dipv6=enabled \
  -Dsqlite=enabled \
  -Dsyslog=enabled \
  -Ddocumentation=disabled \
  -Ddsd=false \
  -Dhttpd=false \
  -Dneighbor=false \
  -Drecorder=false \
  -Dtest=false \
  -Dwave_encoder=true \
  -Dadplug=disabled \
  -Dao=disabled \
  -Daudiofile=disabled \
  -Dbzip2=disabled \
  -Dcdio_paranoia=disabled \
  -Dchromaprint=disabled \
  -Dcurl=disabled \
  -Ddbus=disabled \
  -Dexpat=disabled \
  -Dfaad=disabled \
  -Dflac=disabled \
  -Dfluidsynth=disabled \
  -Dgme=disabled \
  -Diconv=disabled \
  -Dicu=disabled \
  -Did3tag=disabled \
  -Diso9660=disabled \
  -Djack=disabled \
  -Dlame=disabled \
  -Dlibmpdclient=disabled \
  -Dlibsamplerate=disabled \
  -Dmad=disabled \
  -Dmikmod=disabled \
  -Dmms=disabled \
  -Dmodplug=disabled \
  -Dmpcdec=disabled \
  -Dmpg123=disabled \
  -Dnfs=disabled \
  -Dopenal=disabled \
  -Dopus=disabled \
  -Doss=disabled \
  -Dpcre=disabled \
  -Dpulse=disabled \
  -Dqobuz=disabled \
  -Dshine=disabled \
  -Dshout=disabled \
  -Dsidplay=disabled \
  -Dsmbclient=disabled \
  -Dsndfile=disabled \
  -Dsndio=disabled \
  -Dsolaris_output=disabled \
  -Dsoundcloud=disabled \
  -Dsoxr=disabled \
  -Dsystemd=auto \
  -Dtremor=disabled \
  -Dtwolame=disabled \
  -Dudisks=disabled \
  -Dupnp=disabled \
  -Dvorbis=disabled \
  -Dvorbisenc=disabled \
  -Dwavpack=disabled \
  -Dwebdav=disabled \
  -Dwildmidi=disabled \
  -Dyajl=disabled \
  -Dzeroconf=disabled \
  -Dzlib=disabled \
  -Dzzip=disabled
```
but MPD load a lot of libraries at runtime.
Could we reduce the loading?

**Reply by MaxKellermann** [reply]
You havn't posted any data showing which libraries MPD loads.

**Reply by MaxKellermann** [reply]
Also see https://www.musicpd.org/help/

**Comment by lin-silas** [comment]
mpd -V
============================
Music Player Daemon 0.23.5 (0.23.5)
Copyright 2003-2007 Warren Dukes <warren.dukes@gmail.com>
Copyright 2008-2021 Max Kellermann <max.kellermann@gmail.com>
This is free software; see the source for copying conditions.  There is NO
warranty; not even MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Database plugins:
 simple

Storage plugins:
 local


Decoders plugins:
 [ffmpeg] 16sv 3g2 3gp 4xm 8svx aa3 aac ac3 adx afc aif aifc aiff al alaw amr anim apc ape asf atrac au aud avi avm2 avs bap bfi c93 cak cin cmv cpk daud dct divx dts dv dvd dxa eac3 film flac flc fli fll flx flv g726 gsm gxf iss m1v m2v m2t m2ts m4a m4b m4v mad mj2 mjpeg mjpg mka mkv mlp mm mmf mov mp+ mp1 mp2 mp3 mp4 mpc mpeg mpg mpga mpp mpu mve mvi mxf nc nsv nut nuv oga ogm ogv ogx oma ogg omg opus psp pva qcp qt r3d ra ram rl2 rm rmvb roq rpl rvc shn smk snd sol son spx str swf tak tgi tgq tgv thp ts tsp tta xa xvid uv uv2 vb vid vob voc vp6 vmd wav webm wma wmv wsaud wsvga wv wve rtp:// rtsp:// rtsps://
 [pcm]

Filters:


Tag plugins:


Output plugins:
 null fifo pipe alsa snapcast

Encoder plugins:
 null wave

Input plugins:
 file alsa ffmpeg

Playlist plugins:
 extm3u m3u pls cue embcue

Protocols:
 file:// alsa:// ftp:// gopher:// hls+http:// hls+https:// http:// https:// mmsh:// mmst:// rtmp:// rtmps:// rtmpt:// rtmpts:// rtp:// rtsp:// rtsps:// sftp:// srtp://

Other features:
 epoll inotify ipv6 systemd tcp un
==========================

And these libraries will load at runtime ( not complete)
==========================
libOpenCL.so.1        libgsm.so.1           libudev.so.1
libX11.so.6           liblz4.so.1           libva-drm.so.2
libaom.so.0           liblzma.so.5          libva-x11.so.2
libavcodec.so.58      libmp3lame.so.0       libva.so.2
libavformat.so.58     libopenjp2.so.7       libvdpau.so.1
libavutil.so.56       libopenmpt.so.0       libvorbisenc.so.2
libbluray.so.2        librabbitmq.so.4      libvpx.so.6
libbz2.so.1.0         librsvg-2.so.2        libwavpack.so.1
libcairo.so.2         libshine.so.3         libwebp.so.6
libchromaprint.so.1   libsnappy.so.1        libwebpmux.so.3
libcodec2.so.0.9      libspeex.so.1         libx264.so.160
libdav1d.so.4         libsqlite3.so.0       libx265.so.192
libdrm.so.2           libsrt-gnutls.so.1.4  libxml2.so.2
libfmt.so.7           libssh-gcrypt.so.4    libxvidcore.so.4
libgcrypt.so.20       libswresample.so.3    libzmq.so.5
libglib-2.0.so.0      libsystemd.so.0       libzstd.so.1
libgme.so.0           libtheoradec.so.1     libzvbi.so.0
libgnutls.so.30       libtheoraenc.so.1
libgobject-2.0.so.0   libtwolame.so.0
=====================================

I think libwavepack.so.1  libshine.so.3  libmp3lame.so.0 should not show up because I've diabled they.

**Reply by MaxKellermann** [reply]
There's something wrong with your list. Are you sure those are the libraries MPD depends on, or are these all indirect dependencies? Do you know the difference?

**Reply by MaxKellermann** [reply]
For example, MPD would never use libX11, or ZMQ, or libgobject. Your list is bogus!

**Reply by lin-silas** [reply]
That's why I found it weird!

**Comment by leigh123linux** [comment]
Why do you expect less runtime deps?, most of those you list are from ffmpeg libs.

**Reply by lin-silas** [reply]
Thanks, I see.

**Reply by MaxKellermann** [reply]
FFmpeg is also my theory, but unfortunately @lin-silas did not post enough data to back this up.
If you want a MPD with light dependencies, don't use an extremely heavy library like FFmpeg. Or compile a custom FFmpeg without all those useless video/image codecs. FFmpeg is very fine as a catch-all for all those obscure codecs it supports (if you need them), but if you want MPD with Vorbis/MP3/FLAC support, enable 3 decoders for exactly those codecs, and keep FFmpeg disabled.

**Reply by lin-silas** [reply]
Thanks a lot, your suggestions are very helpful.
